config/main.py

```diff
@@ -570,6 +570,82 @@ def is_ipaddress(val):
         return False
     return True
 
+#
+# Check if an interface_name is in a vlan
+#
+def  interface_is_in_vlan(vlan_member_table, interface_name):
+
+    for k,v in vlan_member_table:
+        if v == interface_name:
+            return True
+
+    return False
+
+
+#
+# Check if port is already configured as mirror destination port
+#
+def interface_is_mirror_dst_port(config_db, interface_name):
+    mirror_table = config_db.get_table('MIRROR_SESSION')
+    for k,v in mirror_table.items():
+        if 'dst_port' in v and v['dst_port'] == interface_name:
+            return True
+
+    return False
+#
+# Check if port is already configured with mirror config
+#
+def interface_has_mirror_config(mirror_table, interface_name):
+    for k,v in mirror_table.items():
+        if 'src_port' in v and v['src_port'] == interface_name:
+            return True
+        if 'dst_port' in v and v['dst_port'] == interface_name:
+            return True
+
+    return False
+
+#
+# Check if SPAN mirror-session config is valid.
+#
+def validate_mirror_session_config(config_db, session_name, dst_port, src_port, direction):
+    if len(config_db.get_entry('MIRROR_SESSION', session_name)) != 0:
+        click.echo("Error: {} already exists".format(session_name))
+        return False
+
+    vlan_member_table = config_db.get_table('VLAN_MEMBER')
+    mirror_table = config_db.get_table('MIRROR_SESSION')
+
+    if dst_port is not None:
+        if interface_name_is_valid(dst_port) is False:
+            click.echo("Error: Destination Interface {} is invalid".format(dst_port))
+            return False
+
+        if interface_is_in_vlan(vlan_member_table, dst_port):
+            click.echo("Error: Destination Interface {} has vlan config".format(dst_port))
+            return False
+
+        if interface_has_mirror_config(mirror_table, dst_port):
+            click.echo("Error: Destination Interface {} already has mirror config".format(dst_port))
+            return False
+
+    if src_port is not None:
+        for port in src_port.split(","):
+            if interface_name_is_valid(port) is False:
+                click.echo("Error: Source Interface {} is invalid".format(port))
+                return False
+            if dst_port is not None and dst_port == port:
+                click.echo("Error: Destination Interface cant be same as Source Interface")
+                return False
+            if interface_has_mirror_config(mirror_table, port):
+                click.echo("Error: Source Interface {} already has mirror config".format(port))
+                return False
+
+    if direction is not None:
+        if not any ( [direction == 'rx', direction == 'tx', direction == 'both'] ):
+            click.echo("Error: Direction {} is invalid".format(direction))
```

Comment:
Since direction is optional, can we set the default direction in this??

Reply:
When source port is specified, we can make direction default as both. Let me push that change.

Reply:
Done. Taken care of this.

Reply:
Where is this default handled? I think I missed it.

Reply:
When the src_port is set and direction is not set, we are setting it to both.

Reply:
I see now, thanks!